core/src/main/scala/akka/kafka/KafkaConsumerActor.scala

```diff
@@ -111,29 +117,41 @@ private[kafka] class KafkaConsumerActor[K, V](settings: ConsumerSettings[K, V])
           else reply ! Committed(offsets.asScala.toMap)
         }
       })
-      //right now we can not store commits in consumer - https://issues.apache.org/jira/browse/KAFKA-3412
-      poll()
+      // When many requestors, e.g. many partitions with committablePartitionedSource the
+      // performance is much by collecting more requests/commits before performing the poll.
```

Comment:
It's because how Kafka client works. Poll() is the main communication call which has to be repeated. Commit won't be sent until poll is called.